From 40f9576873a8747b958fb3b4e70fb85165f90938 Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Fri, 5 Oct 2018 19:43:07 +0530
Subject: [PATCH 0460/1852] v4l: xilinx-multi-scaler: Fix pointers in set chk
 chan_stream

commit f438e98c641ba74c9e1c63069ac5275338848568 from
https://github.com/Xilinx/linux-xlnx.git

Driver is setting cap pointer for output, which is incorrect
This patch corrects the functionality by taking proper
pointer.

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 751f6b147071..1b4603c2b2b5 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -750,9 +750,9 @@ xm2msc_set_chan_stream(struct xm2msc_chan_ctx *ctx, bool state, int type)
 	u32 *ptr;
 
 	if (type == XM2MSC_CHAN_OUT)
-		ptr = &ctx->xm2msc_dev->cap_streamed_chan;
-	else
 		ptr = &ctx->xm2msc_dev->out_streamed_chan;
+	else
+		ptr = &ctx->xm2msc_dev->cap_streamed_chan;
 
 	spin_lock(&ctx->xm2msc_dev->lock);
 	if (state)
@@ -770,9 +770,9 @@ xm2msc_chk_chan_stream(struct xm2msc_chan_ctx *ctx, int type)
 	int ret;
 
 	if (type == XM2MSC_CHAN_OUT)
-		ptr = &ctx->xm2msc_dev->cap_streamed_chan;
-	else
 		ptr = &ctx->xm2msc_dev->out_streamed_chan;
+	else
+		ptr = &ctx->xm2msc_dev->cap_streamed_chan;
 
 	mutex_lock(&ctx->xm2msc_dev->mutex);
 	ret = xm2msc_testbit(ctx->num, ptr);
-- 
2.31.1

